doc/conf.py

```diff
@@ -209,51 +198,45 @@
 # directories to ignore when looking for source files.
 exclude_patterns = ["_build", "**.ipynb_checkpoints"]
 
-# The reST default role (used for this markup: `text`) to use for all
-# documents.
-# default_role = None
-
-# If true, '()' will be appended to :func: etc. cross-reference text.
-# add_function_parentheses = True
-
-# If true, the current module name will be prepended to all description
-# unit titles (such as .. function::).
-# add_module_names = True
-
-# If true, sectionauthor and moduleauthor directives will be shown in the
-# output. They are ignored by default.
-# show_authors = False
 
 # The name of the Pygments (syntax highlighting) style to use.
 pygments_style = "sphinx"
 
-# A list of ignored prefixes for module index sorting.
-# modindex_common_prefix = []
-
-# If true, keep warnings as "system message" paragraphs in the built documents.
-# keep_warnings = False
-
 
 # -- Options for HTML output ----------------------------------------------
-
 # The theme to use for HTML and HTML Help pages.  See the documentation for
 # a list of builtin themes.
-html_theme = "sphinx_rtd_theme"
+html_theme = "sphinx_book_theme"
+html_title = ""
+
+html_context = {
+    "github_user": "pydata",
+    "github_repo": "xarray",
+    "github_version": "master",
+    "doc_path": "doc",
+}
 
 # Theme options are theme-specific and customize the look and feel of a theme
 # further.  For a list of options available for each theme, see the
 # documentation.
-html_theme_options = {"logo_only": True}
-
-# Add any paths that contain custom themes here, relative to this directory.
-# html_theme_path = []
-
-# The name for this set of Sphinx documents.  If None, it defaults to
-# "<project> v<release> documentation".
-# html_title = None
+html_theme_options = dict(
+    # analytics_id=''  this is configured in rtfd.io
+    # canonical_url="",
+    repository_url="https://github.com/pydata/xarray",
+    repository_branch="master",
+    path_to_docs="doc",
+    use_edit_page_button=True,
```

Comment:
> Looking at pydata-sphinx-theme and sphinx-book-theme, both seem to have a doc_path or path_to_docs context variable which allows specifying a custom path (I think we can merge this PR in its current state and fix this and any other issues in a new PR, though). 
> The reasoning for the current behavior of the edit-on-github feature was that people were opening PRs to be merged into stable, which would then be dropped on a release. We could avoid that by either always linking to master (but then we can't determine the line numbers; this is what we are doing with scanpydoc), or by always opening a PR which also adds the commit to master (using a bot?)

@keewis, unless I am misunderstanding how the existing `edit on Github` button works, I think these settings allow us to replicate the existing behavior i.e. For (1) regular pages, the button links to the document/file in the master branch. For (2) API pages, the button is disabled and the `[source] link` links to the lines of code in the master branch. Could you elaborate on when/how with the new changes, the `edit on GitHub` button will result in pointing people to the wrong branch?

Reply:
the edit on github link will currently point to the same branch / tag / commit the docs were built for (regardless of the page).

The existing fix (always point to `master`) makes it impossible to accurately determine the line numbers for API pages. Since we might even have removed that function I'm starting to think that it might be easier to not redirect at all but rather make the fact that the PR does not point to `master` more visible.

Not sure what the best way to do so is, though. Maybe have a workflow post a comment if the branch is not the dev branch?

Reply:
Thank you for the clarification! 

> Not sure what the best way to do so is, though. Maybe have a workflow post a comment if the branch is not the dev branch?

I presume it'd be okay to merge this PR and then address this in a separate PR in the near future?

Reply:
for sure!